src/coreclr/vm/FrameTypes.h

```diff
@@ -47,6 +47,10 @@ FRAME_TYPE_NAME(DebuggerClassInitMarkFrame)
 FRAME_TYPE_NAME(DebuggerExitFrame)
 FRAME_TYPE_NAME(DebuggerU2MCatchHandlerFrame)
 FRAME_TYPE_NAME(ExceptionFilterFrame)
+#ifdef FEATURE_INTERPRETER
+FRAME_TYPE_NAME(InterpreterEntryFrame)
```

Comment:
Do we need really to have both entry and exit frames? I would expect that one `InterpreterFrame` should be enough. We will know that we have exited the interpreter if we find some other Frame or managed code first on the stack.

Reply:
I was considering that and this seems to work better with the stack walker. We could possibly have just the `InterpreterEntryFrame`, but then upon reaching it, we would immediately need to switch to the underlying interpreter frames and then once we get out of them, hit the `InterpreterEntryFrame` again and copy the transition record to the context to move to the caller of the sequence of interpreted frames. Also, the `InterpreterEntryFrame` doesn't have a direct pointer to the topmost frame in the `InterpExecMethod`, Vlad was concerned about updating that on every interpreter to interpreter call.

Reply:
Are the interpreter -> AOT code calls going to push InterpreterExitFrame?

I think we should be optimizing for making the interpreter -> AOT calls as cheap as possible.

Reply:
Yes, those would need it too. We can pass current thread as a parameter to the InterpMethodFrame and then use it to push that frame without the thread local access overhead.

Reply:
So the purpose of the `InterpreterEntryFrame` is to store the context so can unwind to AOT callers, while the purpose of the `InterpreterExitFrame` is to make it cleaner to access the top most interpreter frame ? Would we always resume execution in the interpreter loop via a try/catch ? So if EH determines that an interpreter method caught the exception, in order to resume in the interpreter it would throw an exception that will get caught at this interpreter exit location ?